lib/r2d2/google_pay_token.rb

```diff
@@ -71,29 +71,29 @@ def verify_and_parse_message_ecv1
 
     def verify_and_parse_message_ecv2
       raise SignatureInvalidError, 'intermediate certificate is expired' if intermediate_key_expired?
-      raise SignatureInvalidError, 'no valid signature of intermediate key' unless root_key_signature_verified?
-      raise SignatureInvalidError, 'signature of signedMessage does not match' unless intermediate_key_signature_verified?
+      raise SignatureInvalidError, 'no valid signature of intermediate key' unless intermediate_key_signature_verified?
+      raise SignatureInvalidError, 'signature of signedMessage does not match' unless root_key_signature_verified?
 
       JSON.parse(signed_message)
     end
 
     ### ECv2 Methods ###
-    def root_key_signature_verified?
+    def intermediate_key_signature_verified?
```

Comment:
I'm not quite sure I understand what's going on here.  This might just be a confusing diff, but it's sort of making it look like you renamed the methods to reverse their names.  (Maybe what happened is you swapped the order of the methods?)

Reply:
Thats a great question! I am fairly certain the methods were incorrectly named. As in, this method was verifying the intermediate key signature and the other method was verifying the root key signature. I could be incorrect, maybe the naming was intended differently, but reading the Google documentation I am pretty confident. Renaming the methods also makes the errors make a bit more sense and it keeps them in the same order (the order of the verification ultimately doesn't matter as long as both happen).

Reply:
Ah, gotcha - that makes sense.   Thanks for clarifying!